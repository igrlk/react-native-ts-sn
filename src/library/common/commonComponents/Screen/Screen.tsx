import React from 'react';
import { View, ScrollView } from 'react-native';

import styles from './screenStyles';

interface ScreenProps {
	style?: { [key: string]: any };
	children: React.ReactNode;
	isCentered?: boolean;
	scrollRef?: React.RefObject<ScrollView>;
}

export default function Screen({
	style: extraStyle,
	children,
	isCentered,
	scrollRef,
}: ScreenProps) {
	return (
		<View style={{ flex: 1 }}>
			<View style={[styles.screen, isCentered && styles.screenCentered, extraStyle]}>
				<View style={styles.screenInner}>
					<ScrollView ref={scrollRef}>{children}</ScrollView>
				</View>
			</View>
		</View>
	);
}

Screen.defaultProps = {
	style: {},
};
