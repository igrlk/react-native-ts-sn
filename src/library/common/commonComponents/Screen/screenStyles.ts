import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
	screen: {
		paddingTop: 40,
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'flex-start',
		flex: 1,
	},
	screenInner: {
		width: 300,
	},
	screenCentered: {
		justifyContent: 'center',
	},
});
