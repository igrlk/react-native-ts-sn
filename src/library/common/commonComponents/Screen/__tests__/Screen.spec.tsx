import React from 'react';
import renderer from 'react-test-renderer';

import Screen from '../Screen';

describe('Screen', () => {
	it('match to snapshot', () => {
		const tree = renderer.create(<Screen isCentered>Some screen content</Screen>).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
