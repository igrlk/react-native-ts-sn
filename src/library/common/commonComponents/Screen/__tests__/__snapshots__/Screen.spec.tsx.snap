// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen match to snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View>
      <RCTScrollView>
        <View>
          Some screen content
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
